### ISSUE TITLE
LaTeX: long object names leave no room for parameters and their signature overflows in margin

### ISSUE THREAD

**Issue by jfbu** [sphinx-doc/sphinx#12723]
### Describe the bug

See picture
![Capture d’écran 2024-08-01 à 11 21 04](https://github.com/user-attachments/assets/d30489c8-2579-4bbe-9cdb-7c8cda600bab)

Much worst examples are encountered and some are given [in a SymPy issue comment](https://github.com/sympy/sympy/pull/26867#discussion_r1696735523).

### How to Reproduce

Document a complex framework...

### Environment Information

```text
Python version:        3.12.3 (v3.12.3:f6650f9ad7, Apr  9 2024, 08:18:48) [Clang 13.0.0 (clang-1300.0.29.30)])
Python implementation: CPython
Sphinx version:        8.1.0+/7875ec862
Docutils version:      0.20.1
Jinja2 version:        3.1.4
Pygments version:      2.18.0
```

### Additional context

We need refined LaTeX macros detecting overflow and possibly displaying signatures with parameters underneath.

**Comment by electric-coder** [comment]
Ideally you'd want to break the line right after the parenthesis `(` and group the arguments and parameters starting on the next line - somewhat [like Python Black does it][1]. Taking things one step further there's this related FR #11401 

[1]: https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html#how-black-wraps-lines

**Comment by jfbu** [comment]
Attaching here [^1] a small reproducer to play with.  It uses DejaVu fonts (and xelatex) which are a bit larger than our default for fonts (GNU FreeFont for xelatex), this increases the drama.

[12723_signatures.zip](https://github.com/user-attachments/files/16475439/12723_signatures.zip)


[^1]: This must always have been the case but I only recently became aware one could not edit previous comments to add images or any attachments.  Or maybe it is my browser.

**Comment by jfbu** [comment]
@electric-coder Thanks I think this output is what you are hinting (sic) at:

![Capture d’écran 2024-08-03 à 11 41 40](https://github.com/user-attachments/assets/3329d803-e499-4e43-b53f-43d62526cd2c)

Turns out I had from overwork :smiley: (read: old age) completely forgotten that we have already implemented that during the last two years: #11011 and #11444 and perhaps other patches as well.  I participated into reviewing the LaTeX aspects but since then I had completely forgotten it.

The above output is from adding
```
maximum_signature_line_length = 50
```
in conf.py.  I will need to experiment a bit to see if this is ok for [SymPy](https://github.com/sympy/sympy) with the DejaVu fonts.  One aspect though is that it modifies the HTML output as well (for the better, I would say).

I will thus close this issue (it does not look worth it to ask for a setting applying only to LaTeX output).
